### ISSUE TITLE
Add support for compressed optical disk images

### ISSUE THREAD

**Issue by nattthebear** [TASVideos/BizHawk#2046]
These are nifty things.  PCSX2 supports some, as well as other emulators presumably.  I wanted to add this some time ago, but was told not to.

**Comment by Spikestuff** [comment]
"Nifty things"
You mean the "beautiful" loads that are different from a clean image of the game?
You mean, even though [we have the perfect example as to avoid this](http://tasvideos.org/forum/viewtopic.php?t=21609), you want to add something for the general community even though that it's not the focus of the TASing community?

If this somehow screws with loads compared to a clean image then damn on this implementation.

**Comment by nattthebear** [comment]
> You mean the "beautiful" loads that are different from a clean image of the game?

No, this is not about GCN images with parts missing or any similar capability.  This is about bin files compressed with something like gz, with the emulator building an index file automatically and decompressing on the fly.

see also #2245

**Comment by CasualPokePlayer** [comment]
Kind of a dupe with https://github.com/TASEmulators/BizHawk/issues/1482?

**Comment by nattthebear** [comment]
I'm not sure; are choade files compressed?

**Comment by CasualPokePlayer** [comment]
They are (also the most popular generic disk compression format).
